pkg/crawler/crawler.go

```diff
@@ -262,3 +313,230 @@ func (c *Crawler) fetchSHA1(url string) ([]byte, error) {
 	}
 	return sha1b, nil
 }
+
+func (c *Crawler) fetchAndSavePOMLicenseKeys(url string) ([]string, error) {
+	var keys []string
+	resp, err := c.http.Get(url)
+	if resp.StatusCode == http.StatusNotFound {
+		return keys, nil
+	}
+	if err != nil {
+		return keys, xerrors.Errorf("can't get pom xml from %s: %w", url, err)
+	}
+	defer resp.Body.Close()
+
+	var pomProject PomProject
+
+	decoder := xml.NewDecoder(resp.Body)
+	decoder.CharsetReader = charset.NewReaderLabel
+	err = decoder.Decode(&pomProject)
+
+	if err != nil {
+		return keys, xerrors.Errorf("can't parse pom xml from %s: %w", url, err)
+	}
+
+	if len(pomProject.Licenses) == 0 {
+		return keys, nil
+	}
+
+	for _, l := range pomProject.Licenses {
+		key := getLicenseKey(l)
+
+		l.LicenseKey = key
+
+		// update uniqueLicenseKeys map
+		c.uniqueLicenseKeys.Set(key, l)
+
+		keys = append(keys, key)
+	}
+
+	return keys, nil
+
+}
+
+func (c *Crawler) classifyLicense() error {
+	normalizedLicenseMap := make(map[string]string)
+
+	// prepare classifier data i.e create temporary files with license text to be used for classification
+	licenseFiles := c.prepareClassifierData()
```

Comment:
what if we return `filesLicenseMap` here and remove `licenseFiles` slice?

We can use filesLicenseMap.Keys() for Classify:
```
errs := c.classifier.ClassifyLicensesWithContext(ctx, 1000, filesLicenseMap.Keys(), true)
```

Reply:
Valid point :) . I should have noticed this. Fixed

Reply:
Also we can move  filesLicenseMap from Crawler to `classifyLicense` function ( i mean return from `prepareClassifierData` function. We use it only in `classifyLicense` function). 
wdyt?

Reply:
Done